components/ScreenerView/ScreenerViewGeneFilter.vue

```diff
@@ -0,0 +1,360 @@
+
+  export default {
+    components: {
+      VueSimpleSuggest,
+      WarningMessage,
+    },
+    props: {
+      filter: {
+        type: Object,
+        default: () => {},
+      },
+      columns: {
+        type: Array,
+        default: () => [],
+      },
+    },
+    data() {
+      return {
+        isAllChecked: true,
+        datasets,
+        parameters: {
+          text: '',
+        },
+        screenerFilter: this.filter,
+        list: this.filter.list,
+        defaultItem: this.filter.defaultItem,
+      };
+    },
+    computed: {
+      ...mapGetters({
+        activeDataset: 'active_dataset',
+      }),
+      isCheckedSelectedArray() {
+        const filteredList = this.list.filter(({ isChecked }) => isChecked);
+        const isDefaultItem = item => _.isEqual(item, this.defaultItem);
+        const isCheckedSelectedArray = this.list
+          .filter(({ isChecked }) => isChecked)
+          .map(({ isSampleSelected }) => isSampleSelected);
+        for (const [i, item] of filteredList.entries()) {
+          if (isDefaultItem(item) || !item.isChecked) {
+            isCheckedSelectedArray[i] = true;
+          }
+        }
+        return isCheckedSelectedArray;
+      },
+      isAllSampleSelected() {
+        return this.isCheckedSelectedArray.every(Boolean);
+      },
+      groupOptions() {
+        const target = this.activeDataset.dataset;
+        switch (target) {
+          case 'humanFantom5':
+            return this.datasets[0].datasets[0].specificity;
+          case 'gtexV8':
+            return this.datasets[0].datasets[1].specificity;
+          case 'mouseFantom5':
+            return this.datasets[1].datasets[0].specificity;
+          default:
+            return [{ label: 'No useable option found' }];
+        }
+      },
+      resetAllDisabled() {
+        const firstItem = this.list[0];
+        if (_.isEqual(firstItem, this.defaultItem) && this.list.length === 1) {
+          return true;
+        }
+        return false;
+      },
+      getIsAllChecked() {
+        return this.list.every(({ isChecked }) => isChecked);
+      },
+    },
+    watch: {
+      getIsAllChecked(newState) {
+        this.isAllChecked = newState;
+      },
+    },
+    mounted() {
+      this.dispatchAction('INIT');
+    },
+    methods: {
+      autoCompleteStyle(item) {
+        const { isChecked, isSampleSelected } = item;
+        const isDefaultItem = _.isEqual(item, this.defaultItem);
+        if (isChecked && !isSampleSelected && !isDefaultItem)
+          return { defaultInput: 'warning' };
+      },
+      isDisable(item) {
+        return this.list.length <= 1;
+      },
+      isEntropy(className) {
+        return ['emin', 'emax'].includes(className);
+      },
+      dispatchAction(action, index, value) {
+        const targetItem = this.list[index];
+        const defaultItemCopy = { ...this.defaultItem };
+        switch (action) {
+          case 'INIT':
+            this.list.push(defaultItemCopy);
```

Comment:
@maruk0chan How about checking the length of the list at INIT?
This would prevent INIT from pushing and increasing the number of rows every time we return to the project page.

like this
```
case 'INIT':
  if (this.list.length > 0) break;
  this.list.push(defaultItemCopy);
  break;
```